pkg/config/config.go

```diff
@@ -193,6 +193,11 @@ func LoadConfig(configFile string) (Config, error) {
 	if !validRegex {
 		return Config{}, fmt.Errorf("device id regex %q does not contain required regex group %q", cfg.MQTT.DeviceIDRegex.pattern, DeviceIDRegexGroup)
 	}
+
+	if cfg.MQTT.ObjectPerTopicConfig != nil && cfg.MQTT.MetricPerTopicConfig != nil {
+		return Config{}, fmt.Errorf("only one of object_per_topic_config and metric_per_topic_config can be specified")
+	}
+
```

Comment:
I should have said this earlier, but it is worth noting that this is potentially a breaking change for some users. They would have had an incorrect config, but if they were using it to parse json then it would have worked with both configs in there and now it will throw an error. I don't think this is a big deal, but maybe worth noting in the release notes or something.